tests/rust-integration-tests/integration_test/src/tests/seccomp_notify/seccomp_agent.rs

```diff
@@ -24,14 +31,14 @@ pub fn recv_seccomp_listener(seccomp_listener: &Path) -> SeccompAgentResult {
         None,
     )
     .context("failed to create seccomp listener socket")?;
-    socket::bind(socket, &addr).context("failed to bind to seccomp listener socket")?;
+    let socket = socket;
+    socket::bind(socket.as_raw_fd(), &addr).context("failed to bind to seccomp listener socket")?;
     // Force the backlog to be 1 so in the case of an error, only one connection
     // from clients will be waiting.
-    socket::listen(socket, 1).context("failed to listen on seccomp listener")?;
-    let conn = match socket::accept(socket) {
+    socket::listen(&socket.as_fd(), 1).context("failed to listen on seccomp listener")?;
+    let conn = match socket::accept(socket.as_raw_fd()) {
         Ok(conn) => conn,
         Err(e) => {
-            let _ = unistd::close(socket);
             bail!("failed to accept connection: {}", e);
         }
```

Comment:
We should also add a `drop(socket)` here, now that we have removed the manual close? Or we are relying on the destructor to be run when function returns via `bail!` ?

Reply:
I think we can rely on `bail!` to close `socket`, because `bail!` is equivalent to `return`. Moreover, there will likely not be a significant increase in code, so manual `drop()` is not necessary.

Reply:
fair enough :+1: